internal/stream/stream.go

```diff
@@ -94,6 +112,19 @@ func (ps *PushStream) Start() error {
 	return errGrp.Wait()
 }
 
+func (ps *PushStream) restartSubsciber() error {
+	ps.subs.Stop()
+	var err error
+	ps.subs, err = ps.subscriberCore.NewSubscriber(context.Background(), ps.nodeID, ps.subscription, defaultTimeoutMs,
```

Comment:
How is start handled here?

Reply:
inside new Subscriber, core module is running it. https://github.com/razorpay/metro/blob/master/internal/subscriber/core.go#L126

Reply:
The go routine just runs the Start of subscriber, and only when the ctx is done, the subscriber go routine is exited, otherwise it will keep on listening on various channels.